### ISSUE TITLE
sm120,, rtx 5090 help

### ISSUE THREAD

**Issue by WachsenHaus** [cocktailpeanut/fluxgym#355]
/usr/local/lib/python3.10/dist-packages/torch/cuda/__init__.py:235: UserWarning: 
2025-03-03 01:34:41 fluxgym  | NVIDIA GeForce RTX 5090 with CUDA capability sm_120 is not compatible with the current PyTorch installation.
2025-03-03 01:34:41 fluxgym  | The current PyTorch install supports CUDA capabilities sm_50 sm_60 sm_70 sm_75 sm_80 sm_86 sm_90.
2025-03-03 01:34:41 fluxgym  | If you want to use the NVIDIA GeForce RTX 5090 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/
2025-03-03 01:34:41 fluxgym  | 

I'm using docker. Is there any way?

**Comment by ctxadm** [comment]
maybe pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu128

**Comment by WachsenHaus** [comment]
Thanks for the reply. Unfortunately, it says that the library module cannot be found.

I modified the image and download link in the dockerfile.


```
# Base image with CUDA 12.4
FROM nvidia/cuda:12.8.0-base-ubuntu22.04

# Install pip if not already installed
RUN apt-get update -y && apt-get install -y \
    python3-pip \
    python3-dev \
    git \
    build-essential  # Install dependencies for building extensions

# Define environment variables for UID and GID and local timezone
ENV PUID=${PUID:-1000}
ENV PGID=${PGID:-1000}

# Create a group with the specified GID
RUN groupadd -g "${PGID}" appuser
# Create a user with the specified UID and GID
RUN useradd -m -s /bin/sh -u "${PUID}" -g "${PGID}" appuser

WORKDIR /app

# Get sd-scripts from kohya-ss and install them
RUN git clone -b sd3 https://github.com/kohya-ss/sd-scripts && \
    cd sd-scripts && \
    pip install --no-cache-dir -r ./requirements.txt

# Install main application dependencies
COPY ./requirements.txt ./requirements.txt
RUN pip install --no-cache-dir -r ./requirements.txt

# Install Torch, Torchvision, and Torchaudio for CUDA 12.8
#RUN pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu12
RUN pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu128
```

**Comment by ctxadm** [comment]
sure, it was an idea from me ...

have the same problem but on Windows 10 nativ Install with FluxGym...

try the same with ->  pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu128

but i think this could solve the issue: pip install --no-cache-dir -r ./requirements.txt

will try that ...

**Comment by ctxadm** [comment]
can you paste you install config for docker? 

ill try to run docker on windows ...

**Comment by WachsenHaus** [comment]
okay ~~, it is a dockerfile

I simply changed the image and download address.

This is the dockerfile stored in git.

```
# Base image with CUDA 12.4
FROM nvidia/cuda:12.8.0-base-ubuntu22.04

# Install pip if not already installed
RUN apt-get update -y && apt-get install -y \
    python3-pip \
    python3-dev \
    git \
    build-essential  # Install dependencies for building extensions

# Define environment variables for UID and GID and local timezone
ENV PUID=${PUID:-1000}
ENV PGID=${PGID:-1000}

# Create a group with the specified GID
RUN groupadd -g "${PGID}" appuser
# Create a user with the specified UID and GID
RUN useradd -m -s /bin/sh -u "${PUID}" -g "${PGID}" appuser

WORKDIR /app

# Get sd-scripts from kohya-ss and install them
RUN git clone -b sd3 https://github.com/kohya-ss/sd-scripts && \
    cd sd-scripts && \
    pip install --no-cache-dir -r ./requirements.txt

# Install main application dependencies
COPY ./requirements.txt ./requirements.txt
RUN pip install --no-cache-dir -r ./requirements.txt

# Install Torch, Torchvision, and Torchaudio for CUDA 12.8
#RUN pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu12
RUN pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu128

RUN chown -R appuser:appuser /app

# delete redundant requirements.txt and sd-scripts directory within the container
RUN rm -r ./sd-scripts
RUN rm ./requirements.txt

#Run application as non-root
USER appuser

# Copy fluxgym application code
COPY . ./fluxgym

EXPOSE 7860

ENV GRADIO_SERVER_NAME="0.0.0.0"

WORKDIR /app/fluxgym

# Run fluxgym Python application
CMD ["python3", "./app.py"]
```

**Comment by ctxadm** [comment]
ok understood what you meant, ill try that with docker desktop on windows 10 / 22H2

**Comment by ctxadm** [comment]
dont get this running my in config - docker desktop stucks at:  => => # Collecting torch
 => => #   Downloading https://download.pytorch.org/whl/nightly/cu128/torch-2.7.0.dev20250204%2Bcu128-cp310-cp310-ma
 => => # nylinux_2_28_x86_64.whl (1363.8 MB)

its downlaading but doesnt finish, ...tried it multiple times ...

**Comment by WachsenHaus** [comment]
It takes a very long time.

I just used the docker file in the repo here, does it seem wrong?

```
[+] Building 3671.5s (20/20) FINISHED                                                              docker:desktop-linux
 => [fluxgym internal] load build definition from Dockerfile                                                       0.0s
 => => transferring dockerfile: 1.65kB                                                                             0.0s
 => [fluxgym internal] load metadata for docker.io/nvidia/cuda:12.8.0-base-ubuntu22.04                             1.7s
 => [fluxgym internal] load .dockerignore                                                                          0.0s
 => => transferring context: 261B                                                                                  0.0s
 => [fluxgym internal] load build context                                                                          0.0s
 => => transferring context: 2.77kB                                                                                0.0s
 => [fluxgym  1/14] FROM docker.io/nvidia/cuda:12.8.0-base-ubuntu22.04@sha256:12242992c121f6cab0ca11bccbaaf757db8  0.1s
 => => resolve docker.io/nvidia/cuda:12.8.0-base-ubuntu22.04@sha256:12242992c121f6cab0ca11bccbaaf757db893b3065d7d  0.1s
 => CACHED [fluxgym  2/14] RUN apt-get update -y && apt-get install -y     python3-pip     python3-dev     git     0.0s
 => CACHED [fluxgym  3/14] RUN groupadd -g "1000" appuser                                                          0.0s
 => CACHED [fluxgym  4/14] RUN useradd -m -s /bin/sh -u "1000" -g "1000" appuser                                   0.0s
 => CACHED [fluxgym  5/14] WORKDIR /app                                                                            0.0s
 => CACHED [fluxgym  6/14] RUN git clone -b sd3 https://github.com/kohya-ss/sd-scripts &&     cd sd-scripts &&     0.0s
 => CACHED [fluxgym  7/14] COPY ./requirements.txt ./requirements.txt                                              0.0s
 => CACHED [fluxgym  8/14] RUN pip install --no-cache-dir -r ./requirements.txt                                    0.0s
 => [fluxgym  9/14] RUN pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.o  1841.6s
 => [fluxgym 10/14] RUN chown -R appuser:appuser /app                                                              1.8s
 => [fluxgym 11/14] RUN rm -r ./sd-scripts                                                                         0.4s
 => [fluxgym 12/14] RUN rm ./requirements.txt                                                                      0.4s
 => [fluxgym 13/14] COPY . ./fluxgym                                                                               0.2s
 => [fluxgym 14/14] WORKDIR /app/fluxgym                                                                           0.1s
 => [fluxgym] exporting to image                                                                                1824.6s
 => => exporting layers                                                                                         1331.3s
 => => exporting manifest sha256:44d6aa52e69ee2c31d88b6280efc6c16ee85041ab72d3456d566c40178ae6ec1                  0.0s
 => => exporting config sha256:396194fd9fa39fb7024391b736155c347b7851d8b9b9fad62e4208c48d96d0a3                    0.0s
 => => exporting attestation manifest sha256:bde0819d5a43659e2154105bd4eee5e73990b643f01e1c6bc49f6b2479f38b78      0.0s
 => => exporting manifest list sha256:93c8a9118376ab29477212a2b70149573e190d0e6b5ff21271f79b5e43ac3e68             0.0s
 => => naming to docker.io/library/fluxgym:latest                                                                  0.0s
 => => unpacking to docker.io/library/fluxgym:latest                                                             493.1s
 => [fluxgym] resolving provenance for metadata file                                                               0.0s
[+] Running 3/3
```

**Comment by ctxadm** [comment]
you're right, and on a xeon v4 it takes even longer than expected!

Got it running, but connect via http://localhost:7860/ doesnt work for me.

(sry iam not a docker engineer ;))

**Comment by ctxadm** [comment]
do you also get the errors like:

12

ENV PUID=${PUID:-1000}

UndefinedVar: Usage of undefined variable '$PUID'
13

ENV PGID=${PGID:-1000}

UndefinedVar: Usage of undefined variable '$PGID' (did you mean $PUID?)

**Comment by ctxadm** [comment]
C:\Users\ctxusr>docker ps
CONTAINER ID   IMAGE     COMMAND              CREATED          STATUS          PORTS                    NAMES
0f5cc3c875e5   fluxgym   "python3 ./app.py"   21 minutes ago   Up 13 seconds   0.0.0.0:7860->7860/tcp   fluxgym

C:\Users\ctxusr>

i can also connect via telnet on Port 7860,  i think the fluxgym py app is not running.

The question is, how can i start it 

O_o

**Comment by WachsenHaus** [comment]
Unfortunately, it doesn't run with Docker either.

**Comment by WachsenHaus** [comment]
It doesn't work in any environment.

My 5090 is resting now. It's very frustrating.

- Windows does not support CUDA 12.8 version.

- Linux provides CUDA preview version.
- My environment is Windows, so I tried it in the docker environment, but it doesn't work.
- It won't work in Linux OS either. Because it didn't work in docker either.

Basic image creation is now possible with rtx5090 in comfyui.

**Comment by ctxadm** [comment]
Hey Wachsenhaus, i also using ComfyUI for Training and Image Generation with my 5070TI and its working fine so far.

**Comment by WachsenHaus** [comment]
\Yeah I hope fluxgym gets patched soon

**Comment by WachsenHaus** [comment]
> Hey Wachsenhaus, i also using ComfyUI for Training and Image Generation with my 5070TI and its working fine so far.

Are you using fluxgym?

**Comment by fuckingZhang** [comment]
> > > Hey Wachsenhaus, i also using ComfyUI for Training and Image Generation with my 5070TI and its working fine so far.
> > 
> > 
> > Are you using fluxgym?
> 
> NOPE, iam using ComfyUI like i said. Didnt get FluxGym running and are not interessted to waste more time to get it running. Iam using Comfyui also for Training right now on my 5070 ...

can this comfyui train adjust some para like batchsize or learning rate er some other??

**Comment by ctxadm** [comment]
> can this comfyui train adjust some para like batchsize or learning rate er some other??

sure, you can adjust everything!

check the attachement!

[flux_lora_train_example01.json](https://github.com/user-attachments/files/19386518/flux_lora_train_example01.json)

**Comment by WachsenHaus** [comment]
ref . https://github.com/kijai/ComfyUI-FluxTrainer

The dataset folder in the example is based on the root (comfyui) folder. This workflow works well. I am also conducting lora training in comfyui.

---
My environment is rtx5090 and comfyui windows portable version.

For portable version, 
you need to uninstall torch and reinstall it with the latest version.
---
blackwell https://github.com/comfyanonymous/ComfyUI/discussions/6643
---

good luck

**Comment by WachsenHaus** [comment]
triton, my typo

Oh sorry, I just got confused with wan2.1. Triton is not needed for flux trainer.

**Comment by WachsenHaus** [comment]
<img width="813" alt="Image" src="https://github.com/user-attachments/assets/e923eb6e-f241-4037-8581-a1c72ce4c2c3" />

<img width="550" alt="Image" src="https://github.com/user-attachments/assets/668b9ec6-4e60-410c-8405-e5bd987da355" />

It works fine even if you just change the max step. I changed the max step to 500. After about an hour, I got a very good result from Lora.

**Comment by WachsenHaus** [comment]
Sorry, I haven't used flux trainer for a few hours and I'm still learning, so I don't know much...
